sushichef.py

```diff
@@ -0,0 +1,858 @@
+
+
+class ResourceType(object):
+    """
+        Base class for File, WebPage, Video, Audio resources
+    """
+    def __init__(self, type_name=None):
+        LOGGER.info("Resource Type: "+type_name)
+        self.type_name = type_name
+        self.node = None
+
+    def to_file(self, filepath=None):
+        pass
+
+
+class YouTubeResource(ResourceType):
+    def __init__(self, resource_url, type_name="Youtube", lang="en"):
+        super(YouTubeResource, self).__init__(type_name=type_name)
+        self.resource_url = self.clean_url(resource_url)
+        self.file_format = file_formats.MP4
+        self.lang = lang
+
+    def clean_url(self, url):
+        if url[-1] == "/":
+            url = url[:-1]
+        return url.strip()
+
+    @classmethod
+    def is_youtube(self, url, get_channel=False):
+        youtube = url.find("youtube") != -1 or url.find("youtu.be") != -1
+        if get_channel is False:
+            youtube = youtube and url.find("user") == -1 and url.find("/c/") == -1
+        return youtube
+
+    @classmethod
+    def transform_embed(self, url):
+        url = "".join(url.split("?")[:1])
+        return url.replace("embed/", "watch?v=").strip()
+
+    def get_video_info(self):
+        ydl_options = {
+                'writesubtitles': True,
+                'allsubtitles': True,
+                'no_warnings': True,
+                'restrictfilenames':True,
+                'continuedl': True,
+                'quiet': False,
+                'format': "bestvideo[height<={maxheight}][ext=mp4]+bestaudio[ext=m4a]/best[height<={maxheight}][ext=mp4]".format(maxheight='720')
+            }
+
+        with youtube_dl.YoutubeDL(ydl_options) as ydl:
+            try:
+                ydl.add_default_info_extractors()
+                info = ydl.extract_info(self.resource_url, download=False)
+                return info
+            except(youtube_dl.utils.DownloadError, youtube_dl.utils.ContentTooShortError,
+                    youtube_dl.utils.ExtractorError) as e:
+                LOGGER.info('An error occured ' + str(e))
+                LOGGER.info(self.resource_url)
+            except KeyError as e:
+                LOGGER.info(str(e))
+
+    def subtitles_dict(self):
+        subs = []
+        video_info = self.get_video_info()
+        if video_info is not None:
+            video_id = video_info["id"]
+            if 'subtitles' in video_info:
+                subtitles_info = video_info["subtitles"]
+                for language in subtitles_info.keys():
+                    subs.append(dict(file_type=SUBTITLES_FILE, youtube_id=video_id, language=language))
+        return subs
+
+    def process_file(self, download=False, filepath=None):
+        if download is True:
+            video_filepath = self.video_download(download_to=filepath)
+        else:
+            video_filepath = None
+
+        if video_filepath is not None:
+            files = [dict(file_type=content_kinds.VIDEO, path=video_filepath)]
```

Comment:
Please add this line to the dict(
```
ffmpeg_settings={"max_width": 480, "crf": 28}
```
This will convert the video file to hz resolution of `480` and "constant rate factor" of 28 which is some pretty good compression. Ricecooker will do the transcoding on your machine... Watch out CPU will get hot!

This should reduce the overall channel size.